projects/plugins/jetpack/extensions/shared/memberships/settings.js

```diff
@@ -165,52 +177,61 @@ function NewsletterAccessSetupNudge( { stripeConnectUrl, isStripeConnected, hasN
 	}
 }
 
-function NewsletterAccessRadioButtons( {
+export function NewsletterAccessRadioButtons( {
 	onChange,
 	accessLevel,
 	hasNewsletterPlans,
 	stripeConnectUrl,
 	isEditorPanel = false,
+	hasPaywallBlock = false,
+	paywallBlockSettings = false,
 } ) {
 	const isStripeConnected = stripeConnectUrl === null;
 	const instanceId = useInstanceId( NewsletterAccessRadioButtons );
 
 	return (
 		<fieldset className="editor-post-visibility__fieldset">
 			<VisuallyHidden as="legend">{ __( 'Audience', 'jetpack' ) } </VisuallyHidden>
-			{ Object.keys( accessOptions ).map( key => (
-				<div className="editor-post-visibility__choice" key={ key }>
-					<input
-						value={ key }
-						type="radio"
-						checked={ key === accessLevel }
-						className="editor-post-visibility__radio"
-						id={ `editor-post-${ key }-${ instanceId }` }
-						name={ `editor-newsletter-access__setting-${ instanceId }` }
-						aria-describedby={ `editor-post-${ key }-${ instanceId }-description` }
-						disabled={
-							key === accessOptions.paid_subscribers.key &&
-							( ! isStripeConnected || ! hasNewsletterPlans )
-						}
-						onChange={ event => {
-							const obj = {};
-							obj[ META_NAME_FOR_POST_LEVEL_ACCESS_SETTINGS ] = event?.target?.value;
-							return onChange && onChange( obj );
-						} }
-					/>
-					<label
-						htmlFor={ `editor-post-${ key }-${ instanceId }` }
-						className="editor-post-visibility__label"
-					>
-						{ accessOptions[ key ].label }
-					</label>
-					{ key === accessLevel && key !== accessOptions.everybody.key && (
-						<p className="editor-post-visibility__notice">
-							<NewsletterNotice accessLevel={ accessLevel } />
-						</p>
-					) }
-				</div>
-			) ) }
+			{ Object.keys( accessOptions ).map(
+				key =>
+					( ! paywallBlockSettings || key !== accessOptions.everybody.key ) && (
+						<div className="editor-post-visibility__choice" key={ key }>
+							<input
+								value={ key }
+								type="radio"
+								checked={ key === accessLevel }
+								className="editor-post-visibility__radio"
+								id={ `editor-post-${ key }-${ instanceId }` }
+								name={ `editor-newsletter-access__setting-${ instanceId }` }
+								aria-describedby={ `editor-post-${ key }-${ instanceId }-description` }
+								disabled={
+									( key === accessOptions.paid_subscribers.key &&
+										( ! isStripeConnected || ! hasNewsletterPlans ) ) ||
+									( key === accessOptions.everybody.key && hasPaywallBlock )
+								}
```

Comment:
> key === accessOptions.everybody.key && hasPaywallBlock

So for this code, if a Paywall Block exists on the post, we disallow showing the post to everyone. I think I probably missed background convo on this, but I can imagine many situations where a user might want to show the content above the paywall to everyone. Doing so allows you to maximize visibility of the post, and thus possible sales of paid subscriptions.